src/tests.rs

```diff
@@ -0,0 +1,126 @@
+use rusqlite::{ffi, Error, ErrorCode};
+
+use crate::Connection;
+
+#[tokio::test]
+async fn open_in_memory_test() -> Result<(), rusqlite::Error> {
+    let conn = Connection::open_in_memory().await;
+    assert!(conn.is_ok());
+    Ok(())
+}
+
+#[tokio::test]
+async fn call_success_test() -> Result<(), rusqlite::Error> {
+    let conn = Connection::open_in_memory().await?;
+
+    let result = conn
+        .call(|conn| {
+            conn.execute(
+                "CREATE TABLE person(id INTEGER PRIMARY KEY AUTOINCREMENT, name TEXT NOT NULL);",
+                [],
+            )
+        })
+        .await;
+
+    assert_eq!(Ok(0), result);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn call_failure_test() -> Result<(), rusqlite::Error> {
+    let conn = Connection::open_in_memory().await?;
+
+    let result = conn.call(|conn| conn.execute("Invalid sql", [])).await;
+
+    assert_eq!(
+        Err(Error::SqlInputError {
+            error: ffi::Error {
+                code: ErrorCode::Unknown,
+                extended_code: 1
+            },
+            msg: "near \"Invalid\": syntax error".to_string(),
+            sql: "Invalid sql".to_string(),
+            offset: 0
+        }),
+        result
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn close_success_test() -> Result<(), rusqlite::Error> {
+    let conn = Connection::open_in_memory().await?;
+
+    assert!(conn.close().await.is_ok());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn double_close_test() -> Result<(), rusqlite::Error> {
+    let conn = Connection::open_in_memory().await?;
+
+    let conn2 = conn.clone();
+
+    assert!(conn.close().await.is_ok());
+    assert!(conn2.close().await.is_ok());
+
+    Ok(())
+}
+
+#[tokio::test]
+#[should_panic]
+async fn close_call_test() {
+    let conn = Connection::open_in_memory().await.unwrap();
```

Comment:
Currently we `panic` in this case. I'm not sure what to do in a way not to break backward-compatibility.

Reply:
The only idea I have is to add an additional `is_closed -> bool` method, which would return a stored `closed: Arc<AtomicBool>` field value so people can check if `call` will `panic` or not.

Reply:
Another idea:
Create a new `call` method which returns a `Result<R, {our error enum}>`, `our error enum` would then contain a variant to indicate that the `call` could not be performed due to the database being closed.

We can call it `safe_call` or `call_with_result` or something.

Reply:
We should modify the existing `call` method to return our error enum. We should do that in another PR.